sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -366,14 +457,16 @@ def fit(self, X, y, sample_weight=None):
             # shape (n_trees_per_iteration, n_samples) where
             # n_trees_per_iterations is n_classes in multiclass classification,
             # else 1.
-            self._baseline_prediction = self._loss.get_baseline_prediction(
-                y_train, sample_weight_train, self.n_trees_per_iteration_
-            )
+            self._baseline_prediction = np.atleast_1d(
+                self._loss.fit_intercept_only(
+                    y_true=y_train, sample_weight=sample_weight_train
+                )
+            ).T
```

Comment:
I think we can reshape here:

```suggestion
            self._baseline_prediction = self._loss.fit_intercept_only(
                    y_true=y_train, sample_weight=sample_weight_train
                ).reshape((-1, 1))
```

and revert all the `self._baseline_prediction[:, None]`

Reply:
As `_loss.fit_intercept_only` returns a float whenever it is not a multiclass setting, some form of `asarray` or `atleast_1d` is still necessary, right?

Reply:
I think `_loss.fit_intercept_only` returns a `np.float64` or `np.float32`, so the following works:

```python
import numpy as np
np.float64(10).reshape(1, -1)
```

Reply:
I did not know this. Thanks!